trefoil/cli/crs.py

```diff
@@ -0,0 +1,37 @@
+import click
+from netCDF4 import Dataset
+from pyproj import Proj
+
+from trefoil.netcdf import crs
+from trefoil.cli import cli
+
+
+@cli.command(short_help='Set spatial reference information for a variable in an existing dataset')
+@click.argument('filename', type=click.Path(exists=True))
+@click.argument('proj4')
+@click.argument('variables', required=False)
+@click.option('--all', 'all_variables', is_flag=True, default=False, help='Set CRS for all variables in the dataset')
+def set_crs(filename, proj4, variables, all_variables):
+    if not variables and not all_variables:
+        raise click.BadArgumentUsage('No variables specified')
+
+    try:
+        proj = Proj(proj4)
+    except RuntimeError:
+        raise click.BadArgumentUsage('Invalid projection: ' + proj4)
+
+    with Dataset(filename, 'a') as ds:
+        variables_li = []
+        if all_variables:
+            variables_li = list(ds.variables.keys())
```

Comment:
variables should be filtered down to only the data variables (those with at least 2 dimensions).  It doesn't make sense to set a crs on a single dimension variable, such as `years`

Reply:
Good point.